docs/nativeaot.md

```diff
@@ -0,0 +1,61 @@
+# NativeAOT
+
+We've added *experimental* support for using [NativeAOT][1] when publishing iOS,
+tvOS, Mac Catalyst and macOS apps in .NET 8.
+
+NativeAOT may produce smaller and/or faster apps - or it may not. It's very
+important to test and profile to determine the results of enabling NativeAOT.
+
+However, our initial testing shows significant improvements both in size (up
+to 50% smaller) and startup (up to 50% faster). For more information about
+performance see [.NET 8 Performance Improvements in .NET MAUI][3].
+
```

Comment:
This is not exactly confidence inspiring. If I were reading this, I would like to know under what circumstances this would generate larger or slower apps rather than leaving it up to me to profile.

Reply:
Unfortunately there's no easy way to predict how an app will turn out. The fact that projects are trimmed (linked) means that adding a single line of code somewhere can have a tremendous cascading effect, preventing *a lot* of other code from being removed. Then add the effect of different AOT compilers (Mono's AOT might generate bigger code in particular when it comes to generics, but there might be other code patterns that may have the opposite effect) multiplied by whatever code patterns is used by all the code that was brought in by that single line of extra code in the project... the answer really is that each project is different, and that developers must test their projects to see the effect of turning on NativeAOT. I'm not sure how to phrase it in a way that inspires more confidence?

Reply:
@rolfbjarne maybe something like this might help you phrase it in the right way:

```suggestion
Our initial testings shows significant improvements both in size (up
to 50% smaller) and startup (up to 50% faster) compared to the default offering with the Mono runtime (for more information please checkout [.NET 8 Performance Improvements in .NET MAUI][3]). 
However, it is important to point out that NativeAOT requires trimming whole applications `<MtouchLink>Full</MtouchLink>` (enabled by default) and is much more restrictive and aggressive in removing code which is not AOT compatible. As a result, if the applications  is referencing some AOT incompatible constructs (e.g., dynamically referencing code for which a dependency cannot be statically determined), NativeAOT compiler might remove them during build time, which can result with crashes at runtime. To warn users about when this can happen, adequate trimming and AOT warnings are produced and they should not be neglected. Instead, all such warnings should be addressed, code adapted accordingly, application tested and profiled in this configuration. Only then it is safe to assume that the application is compatible with NativeAOT and should not cause any unexpected behaviour. 
Finally, there could be cases when fixing mentioned trimming and AOT warnings is not possible i.e. warnings coming from referenced frameworks or third-party libraries. In such cases, `<MtouchLink>SdkOnly</MtouchLink>` can be specified on the project level as a workaround. This does affect the application size, but even in this mode NativeAOT tends to produce up to 28% smaller applications compared to Mono.
```

Reply:
I updated it according to Ivan's suggestion, @stephen-hawley does it look better now?